### DISCUSSION TITLE
Can't use between portals when users are same SAML login/ID

### DISCUSSION THREAD

**Discussion by GnatRuns** [EsriPS/arcgis-assistant-feedback#37] [Ideas]
I have added/registered the ArcGIS Assistant (AA) app in both my Stage and Production portals (with different AppIDs) and can successfully sign-in to both in AA.  When I am only signed in to one portal (stage) I can list the contents of my users.  However, once I 'refresh credentials' on the other portal (prod), the contents list only reflects the content of last-logged-in portal (prod).  Even though the original portal (stage) is still listed as the "currently active account" the content list only reflects items in the prod portal (the last one to login).  Once this happens, I can't view the content in stage again until I log out of both and login fresh to stage.

NOTE: this issue seems unique to an admin acting on behalf of another user.  We are successful in copying content across portals as long as the SAML user is an admin and is copying their own content.

My functional goal, for example, would be to make Stage my "currently active account", select items owned by a user, and copy those items into the same user's content in the Production portal. Unfortunately, AA appears to list contents based on only the username, instead of the user+portal unique combination.

I suspect this is because both portals use the same Azure Active Directory ID (ex. firstname.lastname.@state.nm.us), even though each portal has a distinct Azure AD application, configured separately for Stage and Prod portals.  (All users in this case are Administrators.)

The workaround involves 1) logging into the source portal (stage) as a non-SAML/local admin, which allows the source user content to be selected for copy.  However, 2) the content copy ends up in the destination admin's content instead of the destination user's content, again presumably because the source/destination username/SAML account is the same in both portals.  Once the content is copied into the destination portal, 3) move the content from the admin's content into the user's content.

**Comment by jpeterson** [comment]
Hey @GnatRuns, thanks for logging this with so much detail. We'll take a look and get back to you with some info.

**Comment by thollingshead** [comment]
Hi @GnatRuns, so far I have been unable to reproduce the behavior you are seeing, although I'm sure my testing setup is differing from yours in some key way.

Would you mind capturing some more info to help debug the behavior? I'd like to see how your Stage and Prod accounts are being stored both before and after the Prod "refresh credentials". You can find this info in your browser's Developer Tools under Local Storage (Chrome example below).

![image](https://user-images.githubusercontent.com/5472848/149600611-0558575f-b751-403f-bac8-623459b3d088.png)
